android/src/com/unciv/app/NativeFontAndroid.kt

```diff
@@ -64,13 +82,35 @@ class NativeFontAndroid(private val size: Int, private val fontFamily: String) :
         return pixmap
     }
 
-    override fun getAvailableFont(): Collection<FontData> {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            SystemFonts.getAvailableFonts().asSequence().mapNotNull {
-                it.file?.nameWithoutExtension
-            }.map { FontData(it) }.toSet()
-        } else {
-            listOf(FontData("sans-serif"), FontData("serif"), FontData("mono"))
+    override fun getAvailableFontFamilies(): Sequence<FontFamilyData> {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q)
+            return sequenceOf(FontFamilyData("sans-serif"), FontFamilyData("serif"), FontFamilyData("mono"))
+
+        fun String.stripFromFirstDash(): String {
+            val dashPos = indexOf('-')
+            if (dashPos < 0) return this
+            return this.substring(0, dashPos)
         }
+
+        // To get _all_ Languages a user has in their Android settings, we would need more help
+        // from the launcher: (Activity).resources.configuration.locales
+        val languageTag = Locale.getDefault().toLanguageTag()  // e.g. he-IL, corresponds to the _first_ Language in Android settings 
+        val supportedLocales = arrayOf(languageTag, "en-US")
+        val supportedLanguages = supportedLocales.map { it.take(2) }
```

Comment:
I'm confused. What does this line achieve?

Reply:
String.take(2) is a non-throwing substring(0,2) to make "en-US" -> "en", and somewhere I stumbled over a string shorter than 2 chars, so the substring threw the Exception, but I don't remember the details. May have been elsewhere and I just stored `String.take` as equivalent to .net Left() in my brainbox.